### PR TITLE
Fix merge table function with empty column set

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Fix no columns error when no columns are used in merge table function. This fixes https://clickhouse-test-reports.s3.yandex.net/21414/789818b59f2155a7c2c50d1255f029c2fd7c0841/fuzzer_debug/report.html#fail1

Detailed description / Documentation draft:

.

### CODE REVIEW

File: src/Storages/StorageMerge.cpp [hunk 1]
```diff
@@ -216,6 +216,10 @@ Pipe StorageMerge::read(
     /// What will be result structure depending on query processed stage in source tables?
     Block header = getHeaderForProcessingStage(*this, column_names, metadata_snapshot, query_info, context, processed_stage);
 
+    /// There must be at least one column in the block so that it knows the number of rows.
+    if (!header)
+        header.insert({ColumnConst::create(ColumnUInt8::create(1, 0), 1), std::make_shared<DataTypeUInt8>(), "_dummy"});
```
Comment (vitlibar) [inline src/Storages/StorageMerge.cpp:221]: It doesn't look quite a good fix, and it also seems to cause the test `00022_merge_prewhere` to fail.

Comment (amosbird) [reply src/Storages/StorageMerge.cpp:221]: +1.

However, we already rely on this fix in other place https://github.com/ClickHouse/ClickHouse/blob/master/src/Interpreters/evaluateConstantExpression.cpp#L47

**Comment by alexey-milovidov** [comment]
@mergifyio update

**Comment by alexey-milovidov** [comment]
FYI #25801

**Pull request closed without merging**